refactor(e2e-tests): import the instantsearch-e2e-tests repo (#5169)
algolia/instantsearch.js: tests/e2e/CONTRIBUTING.md

> # Contributing

Reviewer: Do we still need a contributing guide?

Author: yes, it explains how to write tests

Reviewer: That should be in our [main contributing guide](https://github.com/algolia/instantsearch.js/blob/master/CONTRIBUTING.md#tests) then.

    expect(hitsTitles).toEqual([
      'Nest - Learning Thermostat - 3rd Generation - Stainless Steel',
      'LG - 1.1 Cu. Ft. Mid-Size Microwave - Stainless-Steel',
      'Insignia™ - 2.6 Cu. Ft. Compact Refrigerator - Black',
      'Keurig - K50 Coffeemaker - Black',
      'iRobot - Roomba 650 Vacuuming Robot - Black',
      'Shark - Navigator Lift-Away Deluxe Bagless Upright Vacuum - Blue',
      'LG - 1.5 Cu. Ft. Mid-Size Microwave - Stainless Steel',
      'Insignia™ - 5.0 Cu. Ft. Chest Freezer - White',
      'Samsung - activewash 4.8 Cu. Ft. 11-Cycle High-Efficiency Top-Loading Washer - White',
      'Samsung - 4.8 Cu. Ft. 11-Cycle High-Efficiency Top-Loading Washer - White',
      'LG - 2.0 Cu. Ft. Full-Size Microwave - Stainless Steel',
      'Shark - Rotator Professional Lift-Away HEPA Bagless 2-in-1 Upright Vacuum - Red',
      'LG - 4.5 Cu. Ft. 8-Cycle High-Efficiency Top-Loading Washer - White',
      'Samsung - 4.2 Cu. Ft. 9-Cycle High-Efficiency Steam Front-Loading Washer - Platinum',
      'Panasonic - 1.3 Cu. Ft. Mid-Size Microwave - Stainless steel/black/silver',
      'LG - 2.0 Cu. Ft. Mid-Size Microwave - Black Stainless',
    ]);
  });
});
```

> General guidelines when writing tests:

Reviewer: ```suggestion
Here are some general guidelines when writing end-to-end tests:
```

> - Each step should be translated as a `it` function, while this is not mandatory it helps to make the scenario more readable and properly [separate each actions in Sauce Labs reports](https://user-images.githubusercontent.com/13209/62311104-56217d80-b48b-11e9-94dc-3c18b9ddc2af.png).

Reviewer: ```suggestion
- Each separate step should be in its own `it` function. This isn't a hard rule, but it helps make the scenario more readable and properly [separate each action in Sauce Labs reports](https://user-images.githubusercontent.com/13209/62311104-56217d80-b48b-11e9-94dc-3c18b9ddc2af.png).
```

- All actions on the browser are asynchronous, so be sure to always `await` them. **Never run multiple asynchronous commands in parallel as it can confuse some browsers (Internet Explorer)**.

> - Use helper functions when possible, for readability but also for maintainability (if one widget is updated, we only have to updates its helpers without touching the tests). [More about Helpers](#helpers).

Reviewer: ```suggestion
- Use [helper](#helpers) functions whenever possible, for readability but also for maintainability—if one widget is updated, we only have to updates its helpers without touching the tests.
```

- Do not make assertions to know if an action was correctly performed in the browser. If an action fails (trying to click on an non-existing element for example) then WebdriverIO will automatically throw and fail the test, so asserting on it ourselves is redundant.

> - Only assert what you want to see on the page after an action (Is this checkbox selected? Is the result list correct? etc.)

Reviewer: ```suggestion
- Only assert what you want to see on the page after an action—for example, is the checkbox selected, is the result list correct, etc.
```

> - You may need to add some additional steps compared to the original scenario, to wait for some elements to update for example (this was not done in the example above for simplicity but you can find some examples [here](https://github.com/algolia/instantsearch-e2e-tests/blob/5a2456b8d63afa684931b0447f00df821b02752b/specs/brand-and-query.spec.ts#L14-L16) or [here](https://github.com/algolia/instantsearch-e2e-tests/blob/5a2456b8d63afa684931b0447f00df821b02752b/specs/price-range.spec.ts#L13-L22)).

Reviewer: The links are outdated.
